base/src/test/scala/co/uproot/abandon/DateConstraintTest.scala

```diff
@@ -0,0 +1,67 @@
+package co.uproot.abandon
+
+import java.time.{LocalDate, Month}
+
+import org.scalamock.scalatest.MockFactory
+import org.scalatest._
+
+class DateConstraintTest extends FlatSpec with Matchers with BeforeAndAfterEach with MockFactory with OneInstancePerTest {
+
+  class MockableDetailedPost extends DetailedPost(new AccountName(Seq()), 0, None, None)
+
+  var appState: AppState = null
+
+  override def beforeEach() {
+    setupMocks
+  }
+
+  it should "return true on valid posts" in {
+    val constraint = new DateConstraint(
+      Some(LocalDate.of(2013, Month.JANUARY, 1)),
+      Some(LocalDate.of(2013, Month.DECEMBER, 31))
+    )
+
+    constraint.check(appState) should be(true)
+  }
+
+  it should "throw exception on end date violation" in {
+    val constraint = new DateConstraint(
+      Some(LocalDate.of(2013, Month.JANUARY, 1)),
+      Some(LocalDate.of(2013, Month.NOVEMBER, 1))
+    )
+    an [ConstraintError] should be thrownBy constraint.check(appState)
+
+    val constraintWithoutFrom = new DateConstraint(None, Some(LocalDate.of(2013, Month.NOVEMBER, 1)))
+    an [ConstraintError] should be thrownBy constraintWithoutFrom.check(appState)
+  }
+
+  it should "throw exception on start date violation" in {
+    val constraint = new DateConstraint(
+      Some(LocalDate.of(2013, Month.JUNE, 1)),
+      Some(LocalDate.of(2013, Month.NOVEMBER, 1))
+    )
+    an [ConstraintError] should be thrownBy constraint.check(appState)
+
+    val constraintWithoutTo = new DateConstraint(Some(LocalDate.of(2013, Month.NOVEMBER, 1)), None)
+    an [ConstraintError] should be thrownBy constraintWithoutTo.check(appState)
+  }
+
+  it should "return true on empty dates" in {
+    val constraintWithoutTo = new DateConstraint(None, None)
+    constraintWithoutTo.check(appState) should be(true)
+  }
```

Comment:
Ah, you have considered the `None, None` case. My preference is to treat it as an error, but I am not too particular about it.